src/diart/mapping.py

```diff
@@ -13,7 +13,10 @@ def invalid_tensor(self, shape: Union[Tuple, int]) -> np.ndarray:
         return np.ones(shape) * self.invalid_value
 
     def optimal_assignments(self, matrix: np.ndarray) -> List[int]:
-        return list(lsap(matrix, self.maximize)[1])
+        try:
+            return list(lsap(matrix, self.maximize)[1])
+        except ValueError:
+            print(matrix)
```

Comment:
You're right, it is not. Removed it.

Reply:
Well technically, there has still been a reason. I saw, though very rarely, some ValueErrors like this
```python
  File "/home/simon/PycharmProjects/diart/src/diart/blocks/diarization.py", line 236, in __call__
    permuted_seg = self.clustering(seg, emb)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/simon/PycharmProjects/diart/src/diart/blocks/clustering.py", line 212, in __call__
    self.identify(segmentation, embeddings).apply(segmentation.data),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/simon/PycharmProjects/diart/src/diart/blocks/clustering.py", line 164, in identify
    valid_map = dist_map.unmap_threshold(self.delta_new)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/simon/PycharmProjects/diart/src/diart/mapping.py", line 270, in unmap_threshold
    for src, tgt in zip(*self.valid_assignments())
                         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/simon/PycharmProjects/diart/src/diart/mapping.py", line 225, in valid_assignments
    for src, tgt in enumerate(self._raw_optimal_assignments):
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/simon/PycharmProjects/diart/src/diart/mapping.py", line 196, in _raw_optimal_assignments
    self._opt_assignments = self.objective.optimal_assignments(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/simon/PycharmProjects/diart/src/diart/mapping.py", line 16, in optimal_assignments
    return list(lsap(matrix, self.maximize)[1])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: matrix contains invalid numeric entries
Streaming input_device:None ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   
Took 0.040 (+/-0.027) seconds/chunk -- ran 329 times

```

that I wanted to investigate. For some reason, the matrix at `self.mapping_matrix` contains `nan` from time to time. I was thinking about handling by interpolating `nan` with 0, but I'd still like to know the root cause. Any ideas?

Reply:
Otherwise interpolating with `self.objective.invalid_value` might be an option.

Reply:
To give a bit more intel: this originates from the `unmap_speakers` where target_speakers does not contain speaker 0 and source_speakers does not contain speaker 0 or 1, e.g. `source_speakers=[2]`. The `self.mapping_matrix` does sometimes only have a value for `[0][0]` but not for `[1][0]` (so here is `nan`, as is everywhere else). But while all the other values get filled with invalid, `[1][0]` does not.

Reply:
Mm it's not very clear to me. Can you send the full mapping matrix, as well as active speakers, known centroids and delta new?

Reply:
Sure!
The function stack (test.py at the bottom is my custom testing script):
```python
optimal_assignments, mapping.py:20
_raw_optimal_assignments, mapping.py:200
valid_assignments, mapping.py:229
unmap_threshold, mapping.py:274
identify, clustering.py:164
__call__, clustering.py:212
__call__, diarization.py:195
on_next, map.py:37
on_next, autodetachobserver.py:26
_on_next, do.py:48
on_next, autodetachobserver.py:26
on_next, filter.py:34
on_next, autodetachobserver.py:26
inner, concurrency.py:16
on_next, autodetachobserver.py:26
on_next, map.py:41
on_next, autodetachobserver.py:26
on_completed, toiterable.py:24
on_completed, autodetachobserver.py:44
on_completed, autodetachobserver.py:44
_on_completed_core, subject.py:97
on_completed, observer.py:56
on_completed, subject.py:89
on_next, windowwithcount.py:61
on_next, autodetachobserver.py:26
on_next, map.py:41
on_next, autodetachobserver.py:26
on_next, filter.py:34
on_next, autodetachobserver.py:26
on_next, filter.py:34
on_next, autodetachobserver.py:26
on_next, autodetachobserver.py:26
on_next, map.py:41
on_next, autodetachobserver.py:26
_on_next_core, subject.py:62
on_next, observer.py:26
on_next, subject.py:55
read, sources.py:192
__call__, inference.py:230
<module>, test.py:18
```

Here is the matrix
```python
[[7.84130820e-01 6.92833094e-01 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10]
 [           nan            nan 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10]
 [1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10]]
[[7.84130820e-01 6.92833094e-01 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10]
 [           nan            nan 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10]
 [1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10
  1.00000000e+10 1.00000000e+10 1.00000000e+10 1.00000000e+10]]
```

with `mapped_source_speakers = [0,1]` and `mapped_target_speakers = [0,1]` 

`delta_new` is set to 1

`active_centers = {0, 1}`

`blocked_centers = set()`

`inactive_centers = [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]`

Anything else you want? I also got self.centers as a csv

Reply:
thanks for all the info! it doesn't seem normal to me that the matrix contains NaNs. Does the waveform, segmentation output or embeddings contain NaNs?

I'm assuming this is the mapping matrix just before the call to `unmap_threshold()` in [here](https://github.com/juanmc2005/diart/blob/develop/src/diart/blocks/clustering.py#L157-L164). Could you also post the matrix you get before and after `unmap_speakers()`?

Reply:
Actually, this is the matrix I got after unmap_speakers.

Reply:
I sadly can't step back in pdb right now to give you the exact matrix right before the unmap_speakers, what I could do though is to try to reproduce it again but it's quite difficult as such I kinda don't want to loose the stack right now.

Reply:
The whole pyannote model is meant to produce nan [here ](https://github.com/pyannote/pyannote-audio/blob/0b45103cb228a81a9d9d776cca92694cb30ddb41/pyannote/audio/pipelines/speaker_verification.py#L600) and will only fill the masked parts with actual content.

Reply:
ok so that clarifies things. The NaNs are coming from the new model, not `pyannote/embedding`. So basically the problem here is that speaker 1 is considered active according to `tau_active`, but it doesn't pass the threshold of `min_num_frames` [here](https://github.com/pyannote/pyannote-audio/blob/0b45103cb228a81a9d9d776cca92694cb30ddb41/pyannote/audio/pipelines/speaker_verification.py#L604). In this case I think our safest bet is to check for NaN embeddings (those who contain at least 1 NaN) and remove those embeddings from `active_speakers`.

Essentially, we don't recognize those speakers until they reach the minimum number of samples required by the embedding model.

Reply:
this should be implemented in `OnlineSpeakerClustering.identify()`

Reply:
Okay, I gave it a shot. Any opinions?

Reply:
looks good to me!